EncoreTickets.SDK.Tests/Helpers/ApiServiceMockers/MockersForApiService.cs

```diff
@@ -0,0 +1,148 @@
+                    RestResponseFactory.GetSuccessJsonResponse<T>(client, request, responseContent));
+        }
+
+        public void SetupFailedExecution<T>(string responseContent, HttpStatusCode code)
+            where T : class, new()
+        {
+            RestClientWrapperMock
+                .Setup(x => x.Execute<T>(It.IsAny<IRestClient>(), It.IsAny<IRestRequest>()))
+                .Returns((IRestClient client, IRestRequest request) =>
+                    RestResponseFactory.GetFailedJsonResponse<T>(client, request, responseContent, code));
+        }
+
+        public void VerifyExecution<T>(
+            string baseUrl,
+            string resource,
+            Method method,
+            Dictionary<string, object> expectedQueryParameters = null,
+            string bodyInJson = null)
+            where T : class, new()
+        {
+            VerifyExecution<T>(Times.Once(), baseUrl, resource, method, expectedQueryParameters, bodyInJson);
+        }
+
+        public void VerifyExecution<T>(
+            Times times,
+            string baseUrl,
+            string resource,
+            Method method,
+            Dictionary<string, object> expectedQueryParameters = null,
+            string bodyInJson = null)
+            where T : class, new()
+        {
+            RestClientWrapperMock.Verify(
+                x => x.Execute<T>(
+                    It.Is<IRestClient>(client =>
+                        baseUrl.Equals(client.BaseUrl.ToString(), StringComparison.InvariantCultureIgnoreCase)
+                    ),
+                    It.Is<IRestRequest>(request =>
+                        request.Method == method &&
+                        request.Resource.Equals(resource, StringComparison.InvariantCultureIgnoreCase) &&
+                        request.RequestFormat == DataFormat.Json &&
+                        AreQueryParametersInRequest(request, expectedQueryParameters) &&
+                        IsJsonBodyInRequest(request, bodyInJson))
+                ), times);
+        }
+
+        private Mock<RestClientWrapper> GetRestClientWrapperMock()
+        {
+            return new Mock<RestClientWrapper>();
+        }
+
+        private Mock<ApiRestClientBuilder> GetApiRestClientBuilderMock(Mock<RestClientWrapper> restClientWrapperMock)
+        {
+            var restClientBuilderMock = new Mock<ApiRestClientBuilder>();
+            restClientBuilderMock
+                .Setup(x => x.CreateClientWrapper(It.IsAny<ApiContext>()))
+                .Returns(() => restClientWrapperMock.Object);
+            return restClientBuilderMock;
+        }
+
+        private bool AreQueryParametersInRequest(IRestRequest request,
+            Dictionary<string, object> expectedQueryParameters)
+        {
+            var queryParameters = request.Parameters.Where(p => p.Type == ParameterType.QueryString || p.Type == ParameterType.QueryStringWithoutEncode);
+            if (expectedQueryParameters == null)
```

Comment:
If you write ```expectedQueryParameters = expectedQueryParameters ?? new Dictionary<string, object>();``` and move the foreach body to a method called, for example, Foo, then you can replace the whole method starting from this line with ```return expectedQueryParameters.Count == queryParameters.Count() && expectedQueryParameters.All(kvp => Foo(kvp.Key, kvp.Value));```